kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/adapters/ALBookingRMAdapter.java

```diff
@@ -0,0 +1,534 @@
+        } else if (model != null && model.getTemplateId() == 1) {
+            View itemView = LayoutInflater.from(context).inflate(R.layout.al_guest_details_layout, parent, false);
+            return new GuestCountHolder(itemView);
+        } else if (model != null && (model.getTemplateId() == 2 || hotelList != null)) {
+            View itemView = LayoutInflater.from(context).inflate(R.layout.al_rich_message_item, parent, false);
+            return new AlCardRMAdapter.CardViewHolder(itemView);
+        }
+        return null;
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        bindItems(holder, position);
+    }
+
+    @Override
+    public int getItemCount() {
+        if (hotelList != null) {
+            return hotelList.size();
+        } else if (model.getTemplateId() == 1) {
+            return guestList.size();
+        } else if (model.getTemplateId() == 2) {
+            return payloadList != null ? payloadList.size() : 0;
+        } else if (roomList != null) {
+            return roomList.size();
+        } else if (model.getTemplateId() == 5) {
+            return 1;
+        }
+        return 0;
+    }
+
+    //holder classes >>>
+    private class HotelDetailHolder extends RecyclerView.ViewHolder {
+        ImageView productImage;
+        TextView roomTypeTv;
+        TextView noOfGuestTv;
+        TextView priceTv;
+        TextView totalPriceHeaderTv;
+        TextView totalPriceTv;
+        TextView bookAction;
+
+        public HotelDetailHolder(View itemView) {
+            super(itemView);
+
+            productImage = itemView.findViewById(R.id.productImage);
+            roomTypeTv = itemView.findViewById(R.id.roomTypeTv);
+            noOfGuestTv = itemView.findViewById(R.id.noOfGuestsTv);
+            priceTv = itemView.findViewById(R.id.hotelPriceTv);
+            totalPriceHeaderTv = itemView.findViewById(R.id.totalPriceDistTv);
+            totalPriceTv = itemView.findViewById(R.id.totalPriceTv);
+            bookAction = itemView.findViewById(R.id.bookingAction);
+        }
+    }
+
+    private class BookingDetailsHolder extends RecyclerView.ViewHolder {
+        Spinner titleSpinner;
+        EditText firstNameEt;
+        EditText lastNameEt;
+        EditText emailIdEt;
+        EditText contactNumberEt;
+        TextView submitAction;
+
+        public BookingDetailsHolder(View itemView) {
+            super(itemView);
+
+            titleSpinner = itemView.findViewById(R.id.titleSpinner);
+            firstNameEt = itemView.findViewById(R.id.firstNameEt);
+            lastNameEt = itemView.findViewById(R.id.lastNameEt);
+            emailIdEt = itemView.findViewById(R.id.emailIdEt);
+            contactNumberEt = itemView.findViewById(R.id.contactNumberEt);
+            submitAction = itemView.findViewById(R.id.submitDetails);
+
+            titleList = new ArrayList<>();
+            titleList.add("Title *");
```

Comment:
pls check if this needs to go to strings.xml

Reply:
also if this is a fixed list then this can be moved outside constructor

Reply:
done